### ISSUE TITLE
[:bug bug report] "Optix configuration initialization failed!" with large scene after many iterations.

### ISSUE THREAD

**Issue by DoeringChristian** [mitsuba-renderer/mitsuba3#408]
## Summary
I am optimizing a large scene using the PRB integrator with 1 spp.  It worked fine using [this](https://github.com/mitsuba-renderer/mitsuba3/tree/c2285ad85c8d1cf87d6fb474bcff4256df2839db) commit. But after updating to the latest version it brakes after 496 iterations of optimization from 3 viewpoints each. When increasing the samples per pixel the error appears earlier.

The following error message is returned: 
"Optix configuration initialization failed! The OptiX module compilation did not complete successfully. The module's compilation state is: 0x2363"

I could not yet test which PR exactly breaks it.
I have tried it with the latest main branch as well as the 3.1.1 release which both don't work.

## System configuration

System information:

  OS: Fedora release 36 (Thirty Six)
  CPU: AMD Ryzen 7 5800H with Radeon Graphics
  GPU: NVIDIA GeForce RTX 3070 Laptop GPU
  Python: 3.10.8 (main, Nov 14 2022, 00:00:00) [GCC 12.2.1 20220819 (Red Hat 12.2.1-2)]
  NVidia driver: 520.56.06
  CUDA: 11.8.89
  LLVM: 14.0.5

  Dr.Jit: 0.3.2
  Mitsuba: 3.1.1
     Is custom build? False
     Compiled with: GNU 10.2.1
     Variants:
        scalar_rgb
        scalar_spectral
        cuda_ad_rgb
        llvm_ad_rgb

Please ask if you need further information about my setup.
Thanks for your help.

**Comment by Speierers** [comment]
Hi @DoeringChristian ,

This is concerning. Could you provide the scene in question so that I could try to reproduce this bug on our side?

**Comment by DoeringChristian** [comment]
Hi @Speierers,
thanks for your quick reply.
It seems to crash even when using the cornell box scene.
Here the program used:
```python
import mitsuba as mi
import drjit as dr
import matplotlib.pyplot as plt

mi.set_variant("cuda_ad_rgb")

scene = mi.cornell_box()
print(f"{scene=}")
scene["red"] = {
    "type": "principled",
    "base_color": {"type": "rgb", "value": [1.0, 0.0, 0.0]},
}

scene = mi.load_dict(scene)

ref = mi.render(scene, spp=8)

params = mi.traverse(scene)
print(f"{params=}")

key = "red.base_color.value"
params.keep(key)
params[key] = mi.Color3f(0.0)

opt = mi.ad.Adam(params=params, lr=0.05)

for i in range(1000):
    opt[key] = dr.clamp(opt[key], 0.0, 1.0)
    params.update(opt)
    img = mi.render(scene, params, spp=8)

    loss = dr.mean(dr.sqr(img - ref))

    print(f"{i=}, loss={loss[0]}")

    dr.backward(loss)

    opt.step()
```
In this case 1000 steps of course don't have any effect but after 265 steps the error appears.

**Comment by njroussel** [comment]
Hi @DoeringChristian 

Thanks for the reproducer, I just tried and get the same error at the exact same iteration.
I'll take a deeper look at what's happening.

**Comment by wjakob** [comment]
I tried this on the `drjit-backend-refactor` branch -- it runs all 1000 iterations without crashing (on the CI machine).

**Comment by njroussel** [comment]
Makes sense :+1: , the issue was identified and fixed in this PR: https://github.com/mitsuba-renderer/drjit-core/pull/44
(It just hasn't made it's way to Mitsuba yet.)
